src/MongoDB/ReadPreference.c

```diff
@@ -99,7 +118,7 @@ static PHP_METHOD(ReadPreference, __construct)
 				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be <= %" PRId32 ", %" PHONGO_LONG_FORMAT " given", INT32_MAX, maxStalenessSeconds);
 				return;
 			}
-			if (mode == MONGOC_READ_PRIMARY) {
+			if ((Z_TYPE_P(mode) == IS_LONG && Z_LVAL_P(mode) == MONGOC_READ_PRIMARY) || (Z_TYPE_P(mode) == IS_STRING && strcmp(Z_STRVAL_P(mode), "primary") == 0)) {
```

Comment:
Since we already assigned mode here, can we simply use `mongoc_read_prefs_get_mode(intern->read_preference)` instead of duplicating the integer and string comparison?